### PR TITLE
install script improvements

### PR DESCRIPTION
**Pull request by mxyng**

### CODE REVIEW

File: scripts/install.sh [hunk 1]
```diff
@@ -52,22 +48,43 @@ if [ -n "$NEEDS" ]; then
...
+
+    if runnable systemctl >/dev/null; then
+        $SUDO systemctl restart ollama
 
-status "Installing ollama to $BINDIR..."
-$SUDO install -o0 -g0 -m755 -d $BINDIR
-$SUDO install -o0 -g0 -m755 $TEMP_DIR/ollama $BINDIR/ollama
+        timeout 10 sh -c 'while :; do [ "$(curl -s http://127.0.0.1:11434)" = "Ollama is running" ] && break; sleep 0.2; done' \
+            && status 'Ollama service is available at 127.0.0.1:11434' \
+            || true
```
Comment (BruceMacD) [inline scripts/install.sh:65]: What does this last `|| true` do?

File: scripts/install.sh [hunk 2]
```diff
@@ -52,22 +48,43 @@ if [ -n "$NEEDS" ]; then
     exit 1
 fi
 
-status "Downloading ollama..."
-curl --fail --show-error --location --progress-bar -o $TEMP_DIR/ollama "https://ollama.ai/download/ollama-linux-$ARCH"
+TEMP_DIR=$(mktemp -d)
+cleanup() {
```
Comment (BruceMacD) [inline scripts/install.sh:52]:
```suggestion
finish() {
```

This is doing more than cleanup now

**Review by skydiver1980 — APPROVED** [review]

**Comment by ex3ndr** [comment]
Hey! Since there is a bigger PR, may be it could incorporate this small fix for systemd files?

https://github.com/jmorganca/ollama/pull/1320

**Comment by jmorganca** [comment]
Closing due to age however we can revisit if these changes are still applicable!

**Pull request closed without merging**